app/assets/javascripts/custom_annotation_endpoint.js

```diff
@@ -0,0 +1,180 @@
+/*
+ * annotationsList - current list of OA Annotations
+ * dfd - Deferred Object
+ * init()
+ * search(options, successCallback, errorCallback)
+ * create(oaAnnotation, successCallback, errorCallback)
+ * update(oaAnnotation, successCallback, errorCallback)
+ * deleteAnnotation(annotationID, successCallback, errorCallback)
+ *
+ * getAnnotationInOA(endpointAnnotation)
+ * getAnnotationInEndpoint(oaAnnotation)
+ */
+(function($){
+
+  $.Endpoint = function(options) {
+
+    jQuery.extend(this, {
+      dfd:             null,
+      annotationsList: [],  // OA list for Mirador use
+      windowID:        null,
+      eventEmitter:    null
+    }, options);
+
+    this.init();
+  };
+
+  $.Endpoint.prototype = {
+    init: function() {
+      var _this = this;
+      _this.annotationPath = _this.endpoint.replace('s', '');
```

Comment:
The paths themselves are always pluralized.  Perhaps there's something I'm missing (because it apparently works).

```
annotations GET    /annotations(.:format)                               annotations#index {:format=>:json}
            POST   /annotations(.:format)                               annotations#create {:format=>:json}
 annotation GET    /annotations/:id(.:format)                           annotations#show {:format=>:json}
            DELETE /annotations/:id(.:format)                           annotations#destroy {:format=>:json}

```